datastore/defaultimpl.nim

```diff
@@ -37,6 +37,10 @@ proc defaultModifyGetImpl*(
     except CatchableError as err:
       return failure(err)
 
+    if maybeCurrentData == maybeNewData:
```

Comment:
Well technically you should be able to test for this (the absence of a write), and I'd argue you should test for it as otherwise you can't know if your optimization is actually working, no matter how convincing the code looks. :slightly_smiling_face:

I'm wiling to let this pass - against my own best judgement - if you tell me that writing such a test is complicated.

Reply:
I think that skipping writes is not such a crucial feature that would justify having a test that verify if something was called or not called (one of the worst kinds of tests imo), but I've added such test anyway.